packages/editor/src/store/actions.js

```diff
@@ -195,23 +195,38 @@ export const savePost =
 				),
 			content,
 		};
-		dispatch( { type: 'REQUEST_POST_UPDATE_START', options } );
-		await registry
-			.dispatch( coreStore )
-			.saveEntityRecord(
-				'postType',
-				previousRecord.type,
+		let error;
+		try {
+			edits = await applyFilters(
+				'editor.__unstablePreSavePost',
 				edits,
 				options
 			);
+		} catch ( err ) {
+			error = err;
+		}
 
-		let error = registry
-			.select( coreStore )
-			.getLastEntitySaveError(
-				'postType',
-				previousRecord.type,
-				previousRecord.id
-			);
+		dispatch( { type: 'REQUEST_POST_UPDATE_START', options } );
+		if ( ! error ) {
+			error = await registry
+				.dispatch( coreStore )
+				.saveEntityRecord(
```

Comment:
`saveEntityRecord` doesn't return an error, it returns an "updated record" object.

If you want to catch errors thrown by `saveEntityRecord`, you need to use `catch`, the same way how errors from `await applyFilters` are caught.

Note that `saveEntityRecord` throws an error only when it's called with the `throwOnError: true`. Otherwise it only stores the error to state, to be later retrieved by `getLastEntitySaveError`. Therefore, it's correct to handle both cases: first try to catch an error from `saveEntityRecord`, and if nothing is thrown, look at `getLastEntitySaveError`. But it must be done correctly.

Reply:
@jsnajdr -  I switched the error checking approach for `saveEntityRecord` in d5d7bfce716af2a2565052782bd29599ef07e55f. do I need to add more (a check of `getLastEntitySaveError`?) Looking at other uses in the codebase there is a bit of variety.